front/pages/api/w/[wId]/labs/transcripts.ts

```diff
@@ -0,0 +1,153 @@
+
+export type GetLabsTranscriptsConfigurationResponseBody = {
+  configuration: LabsTranscriptsConfigurationResource | null;
+};
+
+async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<
+    WithAPIErrorReponse<GetLabsTranscriptsConfigurationResponseBody>
+  >
+): Promise<void> {
+  const session = await getSession(req, res);
+  const auth = await Authenticator.fromSession(
+    session,
+    req.query.wId as string
+  );
+
+  const owner = auth.workspace();
+  if (!owner) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_not_found",
+        message: "The workspace was not found.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
+      const transcriptsConfigurationGetRes =
+        await LabsTranscriptsConfigurationResource.findByUserIdAndProvider({
+          userId: owner.id,
+          provider: req.query.provider as LabsTranscriptsProviderType,
+        });
+
+      return res
+        .status(200)
+        .json({ configuration: transcriptsConfigurationGetRes });
+
+    // Update
+    case "PATCH":
+      const {
+        agentConfigurationId: patchAgentId,
+        provider: patchProvider,
+        email: emailToNotify,
+        isActive,
+      } = req.body;
+      if (!patchAgentId || !patchProvider) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The `connectionId` and `provider` parameters are required.",
+          },
+        });
+      }
+
+      const transcriptsConfigurationPatchRes =
+        await LabsTranscriptsConfigurationResource.findByUserIdAndProvider({
+          userId: owner.id,
+          provider: patchProvider as LabsTranscriptsProviderType,
+        });
+
+      if (!transcriptsConfigurationPatchRes) {
+        return apiError(req, res, {
+          status_code: 404,
+          api_error: {
+            type: "configuration_not_found",
+            message: "The configuration was not found.",
+          },
+        });
+      }
+
+      await LabsTranscriptsConfigurationResource.setAgentConfigurationId({
+        agentConfigurationId: patchAgentId,
+        provider: patchProvider,
+        userId: owner.id,
+      });
+
+      if (emailToNotify) {
+        await LabsTranscriptsConfigurationResource.setEmailToNotify({
+          emailToNotify,
+          provider: patchProvider,
+          userId: owner.id,
+        });
+      }
+
+      if (isActive !== undefined) {
+        await LabsTranscriptsConfigurationResource.setIsActive({
+          isActive,
+          provider: patchProvider,
+          userId: owner.id,
+        }).then(() => {
+          if (isActive) {
+            void launchRetrieveNewTranscriptsWorkflow({
+              userId: owner.id,
+              providerId: patchProvider,
+            });
+          }
+        });
+      }
```

Comment:
```suggestion
        await LabsTranscriptsConfigurationResource.setIsActive({
          isActive,
          provider: patchProvider,
          userId: owner.id,
        });
        
          if (isActive) {
            void launchRetrieveNewTranscriptsWorkflow({
              userId: owner.id,
              providerId: patchProvider,
            });
          }
      }
```

Reply:
I'd challenge the need of the `void` when starting the workflow.

Reply:
Using await instead as discussed IRL